src/main/kotlin/ecommerce/GlobalExceptionHandler.kt

```diff
@@ -0,0 +1,57 @@
+package ecommerce
+
+import ecommerce.exception.AuthorizationException
+import ecommerce.exception.InternalServerErrorException
+import ecommerce.exception.NotFoundException
+import org.springframework.dao.DataAccessException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+
+@ControllerAdvice
+class GlobalExceptionHandler {
+    @ExceptionHandler(NotFoundException::class)
+    fun handleNotFoundException(e: NotFoundException): ResponseEntity<Void> {
+        println("NotFoundException occurred: " + e.message)
+        return ResponseEntity.notFound().build()
+    }
+
+    @ExceptionHandler(InternalServerErrorException::class)
+    fun handleInternalServerErrorException(e: InternalServerErrorException): ResponseEntity<Void> {
+        println("InternalServerErrorException occurred: " + e.message)
+        return ResponseEntity.internalServerError().build()
+    }
+
+    @ExceptionHandler(DataAccessException::class)
+    fun handleDataAccessException(e: Exception): ResponseEntity<Void> {
+        println("DataAccessException occurred: " + e.message)
+        return ResponseEntity.internalServerError().build()
+    }
+
+    @ExceptionHandler(IllegalStateException::class)
+    fun handlerIllegalStateException(e: Exception): ResponseEntity<Void> {
+        println("IllegalStateException occurred: " + e.message)
+        return ResponseEntity.internalServerError().build()
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleValidationException(e: MethodArgumentNotValidException): ResponseEntity<Map<String, Any>> {
+        val errors =
+            e.bindingResult.fieldErrors.associate { error ->
+                error.field to (error.defaultMessage ?: "Invalid value")
+            }
+        val errorBody = mapOf("errors" to errors)
+        println("MethodArgumentNotValidException occurred: $errorBody")
+        return ResponseEntity.badRequest().body(errorBody)
+    }
+
+    @ExceptionHandler(AuthorizationException::class)
+    fun handleAuthorizationException(e: AuthorizationException): ResponseEntity<Map<String, Any>> {
+        val error = mapOf("authorization" to e.message)
+        val errorBody = mapOf("errors" to error)
+        println("AuthorizationException occurred: $errorBody")
+        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(errorBody)
+    }
```

Comment:
When it comes to response body of `UNAUTHORIZED` response, I prefer not giving it a body with error information, unless we really need it.
This is because we don't want people to find out what's wrong when they fail to breach into our system. I'm curious about your opinion on this :)